rllib/core/testing/torch/bc_module.py

```diff
@@ -85,3 +86,89 @@ def from_model_config(
         }
 
         return cls(**config)
+
+
+class BCTorchRLModuleWithSharedGlobalEncoder(TorchRLModule):
+    """An example of an RLModule that uses an encoder shared with other things.
+
+    For example, we could consider a multi-agent case where for inference each agent
+    needs to know the global state of the environment, as well as the local state of
+    itself. For better representation learning we would like to share the encoder
+    across all the modules. So this module simply accepts the encoder object as its
+    input argument and uses it to encode the global state. The local state is passed
+    through as is. The policy head is then a simple MLP that takes the concatenation of
+    the global and local state as input and outputs the action logits.
+
```

Comment:
very cool! but how does gradients flow through the shared encoder? it will get some gradients from each single agents modules' backward() right?

Reply:
yes, basically the enablement comes from the fact that you have one memory location for encoder, but the encoder is used in several modules. So if the outputs of encoder is used across several modules, it's gradients would be computed w.r.t. all module losses.

Reply:
cool!